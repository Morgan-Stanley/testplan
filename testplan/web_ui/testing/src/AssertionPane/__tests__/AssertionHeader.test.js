import React from "react";
import { render } from "@testing-library/react";
import { StyleSheetTestUtils } from "aphrodite";

import AssertionHeader from "../AssertionHeader";

const props = {
  assertion: {
    utc_time: "2024-12-24T09:35:56.113946+00:00",
    machine_time: "2024-12-24T17:35:56.113951+08:00",
    type: "Equal",
    meta_type: "assertion",
    description: "soda water == soda water",
    line_no: 26,
    category: "DEFAULT",
    flag: "DEFAULT",
    file_path: "/dummy/examples/Driver/Dependency/suites.py",
    passed: true,
    first: "soda water",
    second: "soda water",
    label: "==",
    type_actual: "str",
    type_expected: "str",
  },
  displayPath: false,
  uid: "c260c2af-888a-4853-86b2-82a14f44066e", // generated
  toggleExpand: jest.fn(),
  showStatusIcons: false,
};

describe("AssertionHeader", () => {
  let shallowComponent;

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    shallowComponent = undefined;
  });

  it("shallow renders the correct HTML structure", () => {
    shallowComponent = render(<AssertionHeader {...props} />);
    expect(
      shallowComponent.getByText(props.assertion.description)
    ).toBeDefined();
    expect(shallowComponent.baseElement).toMatchSnapshot();
  });

  it("shallow renders the correct HTML structure with status icons enabled", () => {
    shallowComponent = render(
      <AssertionHeader {...{ ...props, showStatusIcons: true }} />
    );
    expect(shallowComponent.container.querySelector("svg")).not.toBeNull();
    expect(shallowComponent.baseElement).toMatchSnapshot();
  });

  it("shallow renders the correct HTML structure with code context enabled", () => {
    shallowComponent = render(
      <AssertionHeader {...{ ...props, displayPath: true }} />
    );
    expect(
      shallowComponent.getByText(props.assertion.file_path, { exact: false })
    ).toBeDefined();
    expect(shallowComponent.baseElement).toMatchSnapshot();
  });
});
