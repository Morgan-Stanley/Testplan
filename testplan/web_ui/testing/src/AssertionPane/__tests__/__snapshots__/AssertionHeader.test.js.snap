// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssertionHeader shallow renders the correct HTML structure 1`] = `
<CardHeader
  className="cardHeader_1peii7j-o_O-cardHeaderColorPassed_1nxwz58"
  onClick={[MockFunction]}
  tag="div"
>
  <FontAwesomeIcon
    border={false}
    className="icon_erbgzi"
    fixedWidth={false}
    flip={null}
    icon="clock"
    id="tooltip_0"
    inverse={false}
    key="faClock"
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size="sm"
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
  <Tooltip
    autohide={true}
    delay={
      Object {
        "hide": 250,
        "show": 0,
      }
    }
    hideArrow={false}
    isOpen={false}
    placement="bottom"
    placementPrefix="bs-tooltip"
    target="tooltip_0"
    toggle={[Function]}
  >
    Tue, 12 Feb 2019 17:41:42 GMT
  </Tooltip>
  <span>
    <strong />
    (
    Equal
    )
  </span>
</CardHeader>
`;
